lib/FreeRTOS/tasks.c

```diff
@@ -1165,13 +1165,20 @@ static void prvAddNewTaskToReadyList( TCB_t *pxNewTCB )
 			pxTCB = prvGetTCBFromHandle( xTaskToDelete );
 
 			/* Remove task from the ready list. */
-			if( uxListRemove( &( pxTCB->xStateListItem ) ) == ( UBaseType_t ) 0 )
+			if( listIS_CONTAINED_WITHIN( &( pxReadyTasksLists[ ( pxTCB )->uxPriority ] ), &( ( pxTCB )->xStateListItem ) == pdTRUE )
```

Comment:
The definition of ```taskRESET_READY_PRIORITY``` is:

```c
#define taskRESET_READY_PRIORITY( uxPriority )														\
	{																									\
		if( listCURRENT_LIST_LENGTH( &( pxReadyTasksLists[ ( uxPriority ) ] ) ) == ( UBaseType_t ) 0 )	\
		{																								\
			portRESET_READY_PRIORITY( ( uxPriority ), ( uxTopReadyPriority ) );							\
		}																								\
	}
```

It only calls ```portRESET_READY_PRIORITY```, when the ready list corresponding to the ```uxPriority``` is empty.

Reply:
Oh, yes, you are right.
Sorry, I didn't notice the definition of taskRESET_READY_PRIORITY.  I thought it is the same as portRESET_READY_PRIORITY. My mistake.